src/runtime/string.cpp

```diff
@@ -633,6 +635,28 @@ static PRIMFN(prim_filter_term_codes) {
   RETURN(String::alloc(runtime.heap, ss.str()));
 }
 
+static PRIMTYPE(type_hash_str) {
+  return args.size() == 1 && args[0]->unify(Data::typeString) && out->unify(Data::typeString);
+}
+
+static void blake2b(const std::string &str, uint64_t (*out)[4]) {
+  blake2b_state S;
+  blake2b_init(&S, sizeof(*out));
+  blake2b_update(&S, reinterpret_cast<const uint8_t *>(str.data()), str.size());
+  blake2b_final(&S, reinterpret_cast<uint8_t *>(*out), sizeof(*out));
+}
+
+static PRIMFN(prim_hash_str) {
```

Comment:
Can we make this a stdlib function instead of having consumers rely on the `prim` syntax?

Reply:
Yeah I can add that